ui/pages/confirmations/components/confirm/scroll-to-bottom/scroll-to-bottom.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useContext, useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import { I18nContext } from '../../../../../contexts/i18n';
+import {
+  Box,
+  ButtonIcon,
+  ButtonIconSize,
+  IconName,
+} from '../../../../../components/component-library';
+import type { StyleUtilityProps } from '../../../../../components/component-library/box';
+
+import {
+  BackgroundColor,
+  BlockSize,
+  Display,
+  FlexDirection,
+  IconColor,
+  BorderRadius,
+} from '../../../../../helpers/constants/design-system';
+import { useScrollRequired } from '../../../../../hooks/useScrollRequired';
+import { updateCurrentConfirmation } from '../../../../../ducks/confirm/confirm';
+
+interface ContentProps extends StyleUtilityProps {
+  /**
+   * Elements that go in the page content section
+   */
+  children: React.ReactNode | React.ReactNode[];
+}
+
+const ScrollToBottom = ({ children, ...props }: ContentProps) => {
+  const t = useContext(I18nContext);
+  const dispatch = useDispatch();
+
+  const {
+    hasScrolledToBottom,
+    isScrollable,
+    isScrolledToBottom,
+    onScroll,
+    scrollToBottom,
+    ref,
+  } = useScrollRequired([]);
+
+  useEffect(() => {
+    dispatch(
+      updateCurrentConfirmation({
+        isScrollToBottomNeeded: isScrollable && !hasScrolledToBottom,
+      }),
+    );
+  }, [isScrollable, hasScrolledToBottom]);
```

Comment:
Here we are fetching values from hook and saving into state, this does not looks right approach.

1. If value can be easily derived, lets not set it in state, hook can be used to get it.
2. In case a state update is inevitable, lets do it from the hook itself, we do not need to do it from a component.

Reply:
Hi @jpuri, as I was attempting to ask through our DM, what do you think about us creating a useScrollRequiredConfirm hook that relies on the useScrollRequired  hook?

This is related to our other 2 comments above:
- https://github.com/MetaMask/metamask-extension/pull/22349#discussion_r1490118352
- https://github.com/MetaMask/metamask-extension/pull/22349#discussion_r1491963182

Reply:
Hi @jpuri, as I was attempting to ask through our DM last week, what do you think about us creating a useScrollRequiredConfirm hook that relies on the useScrollRequired hook?

This is related to our other 2 comments above:
https://github.com/MetaMask/metamask-extension/pull/22349#discussion_r1490118352
https://github.com/MetaMask/metamask-extension/pull/22349#discussion_r1491963182